### DISCUSSION TITLE
`vercel dev` requires login

### DISCUSSION THREAD

**Discussion by cjdenio** [vercel/vercel#4613] [Ideas] [answered]
I'm sort of curious, why does `vercel dev` require a login? I'd like to run it in a Docker container (that others will also run), and logging in seems like overkill. Thanks!

**Comment by mcsdevv** [comment, marked as answer]
Hey @cjdenio, it requires login because some of the functionality (an example would be environment variables) requires access to the account/team scope.

We're working on multiple improvements for `vercel dev` and will certainly take this into account.

**Reply by cjdenio** [reply]
I actually just discovered that environment variables can be passed down to `vercel dev` :smile: 

I guess my follow-up question would be: how might I log into the Vercel CLI via a token? (or some other headless method)

**Reply by styfle** [reply]
You can use `vc --token TOKENHERE` https://vercel.com/docs/cli#commands/env/global-options

**Reply by cjdenio** [reply]
That's it. Thank you!

**Reply by MerlinMason** [reply]
Do you have any more details about how you got this working? I'm currently trying to setup docker for local development on a Vercel serverless project.

Here's my Dockerfile
```
# Use node alpine as it's a small node image
FROM node:16-alpine


ARG GITHUB_TOKEN
ENV GITHUB_TOKEN=${GITHUB_TOKEN}
ENV VERCEL_TOKEN=${VERCEL_TOKEN}

# Create the directory on the node image
# where our app will live
RUN mkdir -p /app

# Set /app as the working directory
WORKDIR /app

# Copy package.json and yarn.lock
# to the /app working directory
COPY package.json /app
COPY yarn.lock /app
COPY .yarnrc /app
COPY .npmrc /app

# Install dependencies in /app
RUN yarn install --frozen-lockfile

# Install vercel CLI
RUN npm i vercel --location=global

# Copy the rest of our required things into /app
COPY .vercel /app
COPY utils /app
COPY constants /app
COPY api /app

# Ensure port 3005 is accessible to our system
EXPOSE 3005

# Start dev server
CMD vercel dev --confirm --listen 3005 --token VERCEL_TOKEN --scope healiosuk
```

And I'm getting the following error:
> Error! Could not retrieve Project Settings. To link your Project, remove the `.vercel` directory and deploy again.

What's the best practice here? Copy the `.vercel` directory across? Any help appreciated!
